junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -28,6 +29,8 @@
     private final CountDownLatch startable;
     private final CountDownLatch finished;
     private volatile Thread testThread;
+    
+    private final AtomicBoolean exceptionOccurred = new AtomicBoolean(false); 
```

Comment:
`setInterruptOnException` is only setting up the connection to the main thread, the one that should be interrupted. In case there is already an exception in the latch, it will be interrupted. 

The design is not based only on the current implementation of the junit module. `setInterruptOnException` is a public method, it can be called from any place, any time. As such, it should be implemented to work properly for any future change that will call this method from other places, on any other thread.

Related to the actual fix, what I found was that in a test where k3po rule is around the gateway rule, if there was an exception from the gateway statement, then `notifyException` was called twice, once for the actual exception and a second time when an exception was thrown when k3po was trying to clean up.  For me it seems to difficult to make sure that `notifyException` is called only once from the entire flow, so it is much cleaner and simpler just to make sure that the main thread is interrupted only once. Otherwise we will have again (what I have actually see in the logs) interrupted status leaked from one test to the next one.

As for multiple exceptions, maybe we should record all of them, not only the last one, as we do now.